client/client_test.go

```diff
@@ -976,16 +976,20 @@ func TestHTTPClusterClientSyncUnpinEndpoint(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error during setup: %#v", err)
 	}
-	wants := []string{"http://127.0.0.1:2379", "http://127.0.0.1:4001", "http://127.0.0.1:4002"}
+	wants := map[string]struct{}{
+		"http://127.0.0.1:2379": {},
+		"http://127.0.0.1:4001": {},
+		"http://127.0.0.1:4002": {},
+	}
 
 	for i := 0; i < 3; i++ {
 		err = hc.Sync(context.Background())
 		if err != nil {
 			t.Fatalf("#%d: unexpected error during Sync: %#v", i, err)
 		}
-
-		if g := hc.endpoints[hc.pinned]; g.String() != wants[i] {
-			t.Errorf("#%d: pinned endpoint = %v, want %v", i, g, wants[i])
+		g := hc.endpoints[hc.pinned].String()
+		if _, ok := wants[g]; !ok {
```

Comment:
This doesn't do what the test says it does any more; `Sync` could never change endpoints but the test would still pass. The client itself probably needs a fix.